test/test_transaction_public_blocking_api.cxx

```diff
@@ -499,7 +503,6 @@ TEST_CASE("transactions public blocking API: some query errors are seen immediat
       [](couchbase::transactions::attempt_context& ctx) {
           auto [e, res] = ctx.query("I am not a valid n1ql query");
           CHECK(e.ec());
-          CHECK(std::holds_alternative<couchbase::query_error_context>(e.cause()));
```

Comment:
Instead of removing this entirely, we should replace this with a check that `e.cause().has_value()` and/or `e.cause().value() == errc::common::parsing_failure`